### PR TITLE
FIX: Fix error in report when sklearnex has more parameters than sklearn

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Currently, when creating a report comparing sklearn vs. sklearnex, if some algorithm has more parameters in sklearnex  than in sklearn (such as tree-based algorithms like `ExtraTreesClassifier`), an error will occur when trying to generate the excel report from the JSONs, due to being unable to find those sklearnex-specific columns in the data from sklearn.

This PR fixes the issue by dropping all parameter columns that are not present in data from all the compared libraries, but I am not sure that this wouldn't break some other use-case, so please give it a thorough review.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
Looks like this PR doesn't work in all cases. Sometimes it leaves rows where there are times for both sklearn and sklearnex but no comparison metrics (e.g. time improvement) calculated for them.
